Gems/Atom/RHI/Code/Source/RHI/FreeListAllocator.cpp

```diff
@@ -118,11 +118,6 @@ namespace AZ::RHI
 
     VirtualAddress FreeListAllocator::Allocate(size_t byteCount, size_t byteAlignment)
     {
-        if (byteCount == 0)
```

Comment:
Why remove this? It seems like an optimization.

Reply:
Wouldn't the removal of this impact other RHIs? Sounds like this is something we should handle within the Webgpu rhi.